src/commands/node.mjs

```diff
@@ -2316,8 +2362,1051 @@ export class NodeCommand extends BaseCommand {
+        }
+      },
+      {
+        title: 'Update chart to use new configMap',
+        task: async (ctx, task) => {
+          const config = /** @type {NodeUpdateConfigClass} **/ ctx.config
+          const index = config.existingNodeIds.length
+          const nodeId = Templates.nodeNumberFromNodeId(config.nodeId) - 1
+          let valuesArg = ''
+          for (let i = 0; i < index; i++) {
+            if (i !== nodeId) {
+              valuesArg += ` --set "hedera.nodes[${i}].accountId=${config.serviceMap.get(config.existingNodeIds[i]).accountId}" --set "hedera.nodes[${i}].name=${config.existingNodeIds[i]}"`
+            } else {
+              valuesArg += ` --set "hedera.nodes[${i}].accountId=${config.newAccountNumber}" --set "hedera.nodes[${i}].name=${config.existingNodeIds[i]}"`
+            }
+          }
+          this.profileValuesFile = await self.profileManager.prepareValuesForNodeAdd(
+            path.join(config.stagingDir, 'config.txt'),
+            path.join(config.stagingDir, 'templates', 'application.properties'))
+          if (this.profileValuesFile) {
+            valuesArg += this.prepareValuesFiles(this.profileValuesFile)
+          }
+
+          await self.chartManager.upgrade(
+            config.namespace,
+            constants.FULLSTACK_DEPLOYMENT_CHART,
+            config.chartPath,
+            valuesArg,
+            config.fstChartVersion
+          )
+        }
+      },
+      {
+        title: 'Kill nodes to pick up updated configMaps',
+        task: async (ctx, task) => {
+          const config = /** @type {NodeUpdateConfigClass} **/ ctx.config
+          for (const /** @type {NetworkNodeServices} **/ service of config.serviceMap.values()) {
+            await self.k8.kubeClient.deleteNamespacedPod(service.nodePodName, config.namespace, undefined, undefined, 1)
+          }
+        }
+      },
+      {
+        title: 'Check node pods are running',
+        task:
+          async (ctx, task) => {
+            const subTasks = []
+            const config = /** @type {NodeUpdateConfigClass} **/ ctx.config
+            ctx.config.allNodeIds = ctx.config.existingNodeIds
+            // nodes
+            for (const nodeId of config.allNodeIds) {
+              subTasks.push({
+                title: `Check Node: ${chalk.yellow(nodeId)}`,
+                task: () =>
+                  self.k8.waitForPods([constants.POD_PHASE_RUNNING], [
+                    'fullstack.hedera.com/type=network-node',
+                    `fullstack.hedera.com/node-name=${nodeId}`
+                  ], 1, 60 * 15, 1000) // timeout 15 minutes
+              })
+            }
+
+            // set up the sub-tasks
+            return task.newListr(subTasks, {
+              concurrent: false, // no need to run concurrently since if one node is up, the rest should be up by then
+              rendererOptions: {
+                collapseSubtasks: false
+              }
+            })
+          }
+      },
+      {
```

Comment:
prepare staging directory: https://github.com/hashgraph/solo/blob/681d5b63a94324e73aace70d7d4190a4535a129d/src/commands/node.mjs#L1921-L1921 to copy the keys from the cache/keys to the staging directory

Reply:
Updated